src/test/versionbits_tests.cpp

```diff
@@ -214,10 +235,15 @@ BOOST_AUTO_TEST_CASE(versionbits_test)
                            .Mine(3000, TestTime(10000), 0).TestStarted().TestStateSinceHeight(3000)
                            .Mine(4000, TestTime(10000), 0).TestStarted().TestStateSinceHeight(3000)
                            .Mine(5000, TestTime(10000), 0).TestStarted().TestStateSinceHeight(3000)
+                           .Mine(5999, TestTime(20000), 0).TestStarted().TestStateSinceHeight(3000)
                            .Mine(6000, TestTime(20000), 0).TestFailed().TestStateSinceHeight(6000)
-                           .Mine(7000, TestTime(20000), 0x100).TestFailed().TestStateSinceHeight(6000);
+                           .Mine(8000, TestTime(20000), 0x100).TestFailed().TestStateSinceHeight(6000)
```

Comment:
tests pass without this change included fwiw

Reply:
It gives it two periods in FAILED that are signalling instead of one to give it plenty of time to decide to switch to some other state if there's some bug that allows transitions out of FAILED. (If it were to go out of FAILED and back again, the StateSinceHeight check should catch that)

Reply:
Seems reasonable, although it may be worth explicitly commenting that:

```diff
diff --git a/src/test/versionbits_tests.cpp b/src/test/versionbits_tests.cpp
index 5f53da8aff..19bf3f948f 100644
--- a/src/test/versionbits_tests.cpp
+++ b/src/test/versionbits_tests.cpp
@@ -237,7 +237,8 @@ BOOST_AUTO_TEST_CASE(versionbits_test)
                            .Mine(5000, TestTime(10000), 0).TestStarted().TestStateSinceHeight(3000)
                            .Mine(5999, TestTime(20000), 0).TestStarted().TestStateSinceHeight(3000)
                            .Mine(6000, TestTime(20000), 0).TestFailed().TestStateSinceHeight(6000)
-                           .Mine(8000, TestTime(20000), 0x100).TestFailed().TestStateSinceHeight(6000)
+                           .Mine(7000, TestTime(20000), 0x100).TestFailed().TestStateSinceHeight(6000)
+                           .Mine(8000, TestTime(20000), 0x100).TestFailed().TestStateSinceHeight(6000) // Make sure we don't transition out of FAILED
         ;
     }
 }
```

Reply:
Added 7000 back, bumped 8000 to 24000 as in some of the other cases and added a comment.